src/Pact/Types/PactError.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+-- |
+-- Module      :  Pact.Types.PactError
+-- Copyright   :  (C) 2019 Stuart Popejoy, Kadena LLC
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  Stuart Popejoy <stuart@kadena.io>
+--
+-- PactError and related types.
+--
+
+module Pact.Types.PactError
+  ( StackFrame(..), sfName, sfLoc, sfApp
+  , PactError(..)
+  , PactErrorType(..)
+  ) where
+
+
+
+import Control.Lens hiding ((.=),DefName)
+import Control.Monad.Catch
+import Data.Aeson hiding (Object)
+import Data.Default
+import Data.Text (Text, unpack)
+
+import GHC.Generics
+
+import Pact.Types.Lang
+import Pact.Types.Orphans ()
+import Pact.Types.Pretty
+
+data StackFrame = StackFrame {
+      _sfName :: !Text
+    , _sfLoc :: !Info
+    , _sfApp :: Maybe (FunApp,[Text])
+    } deriving (Eq,Generic)
+instance ToJSON StackFrame where toJSON = toJSON . show
+
+instance Show StackFrame where
+    show (StackFrame n i app) = renderInfo i ++ ": " ++ case app of
+      Nothing -> unpack n
+      Just (_,as) -> "(" ++ unpack n ++ concatMap (\a -> " " ++ unpack (asString a)) as ++ ")"
+makeLenses ''StackFrame
```

Comment:
I love this change - I was hoping for an excuse to move all of the error-handling code into its own `.hs` at some point, along with the error functions in `Runtime.hs` (which I recommend adding here and making them generic in their monad). Can we move `Stackframe` to its own file for its own concerns? both `Runtime` and `PactError` depend on it for different reasons, and its placement here makes it another surprising type, like `Gas` in `Term`.

Reply:
I'll leave that for a later pass, check out the shiny `Pact.Types.Capability`